libs/git/clone_test.go

```diff
@@ -0,0 +1,51 @@
+package git
+
+import (
+	"context"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGitExpandUrl(t *testing.T) {
+	// case: only repository name is specified
+	assert.Equal(t, "https://github.com/databricks/abc", expandUrl("abc"))
+
+	// case: both org and repo name are specified
+	assert.Equal(t, "https://github.com/abc/def", expandUrl("abc/def"))
+
+	// case: full path of the repo is specified
+	assert.Equal(t, "https://github.com/databricks/cli.git", expandUrl("https://github.com/databricks/cli.git"))
+	assert.Equal(t, "git@github.com:databricks/cli.git", expandUrl("git@github.com:databricks/cli.git"))
+
+	// case: invalid repository references
+	assert.Equal(t, "https://github.com/databricks/", expandUrl(""))
+	assert.Equal(t, "https://github.com//", expandUrl("/"))
+	assert.Equal(t, "https://github.com//abc/def", expandUrl("/abc/def"))
```

Comment:
These should return errors.

Reply:
removed the expand url function

Reply:
It's complex to have a simple and robust way to parse these URLs, across different protocols, git providers and operating systems. I decided to just support the repository name and the full URL to keep it simpler, with a simple rubrik, that alpha numeric (and other characters ., -, _ , see: https://stackoverflow.com/questions/59081778/rules-for-special-characters-in-github-repository-name) without any `/` will be resolved against databricks

Reply:
Note `^[\w-\.]+/[\w-\.]+$` is likely a valid regex that can be used as a capturing org name and repository name, but that can be added in as a followup